pkg/readiness/tracker_map.go

```diff
@@ -118,3 +131,24 @@ func (t *trackerMap) Populated() bool {
 	}
 	return true
 }
+
+// Populated returns true if all objectTrackers are populated.
+func (t *trackerMap) TryCancel(g schema.GroupVersionKind) bool {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	obj, ok := t.tryCanceled[g]
```

Comment:
what is the purpose of a new `tryCanceled` map?  Aren't retries held on individual instances of `objectTracker`?

Reply:
We need to TryCancel the instance of  a`data` ot entry iiuc, and not a specific instance of an `ot` w gvk G.